src/mo_def/compUnit.ml

```diff
@@ -30,8 +30,16 @@ let comp_unit_of_prog as_lib (prog : prog) : comp_unit =
     match ds with
     (* imports *)
     | {it = LetD ({it = VarP n; _}, ({it = ImportE (url, ri); _} as e)); _} :: ds' ->
-      let i : import = { it = (n, url, ri); note = e.note.note_typ; at = e.at } in
+      let i : import = { it = (Surface n, url, ri); note = e.note.note_typ; at = e.at } in
       go (imports @ [i]) ds'
+    | {it = LetD ({it = ObjP []; _}, _); _} :: ds' ->
+      go imports ds'
+    | ({it = LetD (({it = ObjP (f :: fs); _} as letd), ({it = ImportE (url, ri); _} as es)); _} as imp) :: ds' ->
+      let m, tfs = Type.as_obj es.note.note_typ in
+      assert (m = Type.Module);
+      let field_id = f.it.id in
+      let i : import = { it = (Bulk (field_id, es.note), url, ri); note = Type.lookup_val_field field_id.it tfs; at = es.at } in
```

Comment:
Instead of `Bulk (field_id, es.note)` just capture the entire binding group (`{it = ObjP fs; _}`) and send it to the other side!

Reply:
Nice idea: 0eff5af79de19d96a7985fec74a4957906ed4c45